doc/user-guide/user-guide.rst

```diff
@@ -122,107 +126,262 @@ To deal with this process in a more comprehensive way, this repository uses a di
+      git checkout -b tickets/DM-12345  # create new branch to work on
 
-.. prompt:: bash
 
-  cycle/cycle.env
+  * Open the ``cycle/cycle.env`` file and update the revision number on the top.
+
+    .. important::
+
+      For now we do not have any protection against reusing an existing revision so make sure you update this number before anything else.
+
+      Eventually this will be enforced by the build system.
+
+  * Update the version of any package you need in ``cycle/cycle.env``.
+
+  * Commit and push your changes.
+
+    If you are building an existing revision, just do
+
+    .. prompt:: bash
+
+      git push
+
+    If this is a new revision you may have to do:
+
+    .. prompt:: bash
+
+      git push --set-upstream origin tickets/DM-12345  # update branch name to match your local branch
+
+    At this stage, make sure you are not rebasing or overwriting the git history.
+    In general, the commits should only be cleaned up once the revision is ready to be closed.
+  
+  * After pushing your changes to your branch you can build the components you want to update.
+
+    The build is entirely handled by the `Telescope and Site Software Jenkins server`_.
+
+    .. _Telescope and Site Software Jenkins server: https://tssw-ci.lsst.org
+
+    Once the changes are pushed to GitHub the branch will appear in the `cycle build jenkins job`_.
+    Select the branch you are working on in the Branch tab and select ``Build with Parameters`` on the left-hand side, which will take you to the :ref:`fig-jenkins-build-with-parameters`.
+
+    If you just created the branch (e.g. you are working on a brand new revision, instead of an existing one), you may see a ``Build`` link instead of ``Build with Parameters``.
+    If this is the case, click on ``Build``, wait for a build to start and then reload the page.
+    This will cause Jenkins to perform a "dry run" and update the build page.
+
+    .. _cycle build jenkins job: https://tssw-ci.lsst.org/view/CycleBuild/job/cycleBuild/
+
+    .. figure:: /_static/JenkinsBuildWithParameters.png
+      :name: fig-jenkins-build-with-parameters
+      :target: ../_images/SingleProcessFig.png
+      :alt: Jenkins build with parameters page
+
+      Jenkins build with parameters page.
+
+  * If the component you want to build is part of the base components, select "base_components", then select the component you want to build in the dropdown menu, and press the "Build" button at the bottom of the page.
+
+    When building a revision it is not possible to build "ALL" the components.
+
+    .. figure:: /_static/JenkinsBuild.gif
+      :name: fig-jenkins-build-base-component
+      :target: ../_images/JenkinsBuild.gif
+      :alt: Jenkins build base component
+
+      Animated gif showing how to build a base component.
+    
+  
+  * If your component is not in the list of "base_components", select it from the available options and press the "Build" button at the bottom of the page.
```

Comment:
`not in the list of` -> `not part of` perhaps? If the component is supposed to be a `base_component` but it is not in the list then `ts_cycle_build` was not updated correctly.